server/app/com/xsn/explorer/services/TransactionRPCService.scala

```diff
@@ -114,4 +117,29 @@ class TransactionRPCService @Inject()(
 
     result.toFuture
   }
+
+  def encodeTPOSContract(
+      tposAddressStr: String,
+      merchantAddressStr: String,
+      commission: Int,
+      signature: String
+  ): FutureApplicationResult[JsValue] = {
+    val result = for {
+      tposAddress <- addressValidator.validate(tposAddressStr).toFutureOr
+      merchantAddress <- addressValidator.validate(merchantAddressStr).toFutureOr
+      encodedTposContract <- xsnService
+        .encodeTPOSContract(
+          tposAddress = tposAddress,
+          merchantAddress = merchantAddress,
+          commission = commission,
+          signature = signature
+        )
+        .toFutureOr
+    } yield {
+      logger.info(encodedTposContract)
```

Comment:
maybe some text on this log?, like `logger.info(s"encoded TPoS contract: $encodedTposContrac"t)`

Reply:
it was an unnecessary log, i removed